src/chat-manager.js

```diff
@@ -0,0 +1,62 @@
+import { BaseClient, HOST_BASE, Instance } from 'pusher-platform'
+import { split } from 'ramda'
+
+import { CurrentUser } from './current-user'
+import { typeCheck, typeCheckObj } from './utils'
+
+export class ChatManager {
+  constructor ({ instanceLocator, tokenProvider, userId, ...options } = {}) {
+    typeCheck('instanceLocator', 'string', instanceLocator)
+    typeCheck('tokenProvider', 'object', tokenProvider)
+    typeCheck('tokenProvider.fetchToken', 'function', tokenProvider.fetchToken)
+    typeCheck('userId', 'string', userId)
+    const cluster = split(':', instanceLocator)[1]
+    if (cluster === undefined) {
+      throw new TypeError(
+        `expected instanceLocator to be of the format x:y:z, but was ${instanceLocator}`
+      )
+    }
+    const baseClient = options.baseClient || new BaseClient({
+      host: `${cluster}.${HOST_BASE}`,
+      logger: options.logger
+    })
+    if (typeof tokenProvider.setUserId === 'function') {
+      tokenProvider.setUserId(userId)
+    }
+    const instanceOptions = {
+      client: baseClient,
+      locator: instanceLocator,
+      logger: options.logger,
+      tokenProvider
+    }
+    this.apiInstance = new Instance({
+      serviceName: 'chatkit',
+      serviceVersion: 'v1',
+      ...instanceOptions
+    })
+    this.filesInstance = new Instance({
+      serviceName: 'chatkit_files',
+      serviceVersion: 'v1',
+      ...instanceOptions
+    })
+    this.cursorsInstance = new Instance({
+      serviceName: 'chatkit_cursors',
+      serviceVersion: 'v1',
+      ...instanceOptions
+    })
+    this.userId = userId
+  }
+
+  connect = (hooks = {}) => {
+    typeCheckObj('hooks', 'function', hooks)
+    const currentUser = new CurrentUser({
+      id: this.userId,
+      apiInstance: this.apiInstance
+    })
+    return Promise.all([
+      currentUser.establishUserSubscription(hooks),
+      currentUser.establishPresenceSubscription(hooks)
```

Comment:
Yeah good point, if either fails we'll end up in the catch. This is less of an issue when you can disable presence, but even then... if someone has everything on by default, would it be better to go on even if the presence subscription fails on the off chance the client isn't using any of the presence stuff? If they are it would break stuff later on, but maybe partial functionality is better than no functionality?

It wouldn't be a big change, we could just have `establishPresenceSubscription` always resolve, and the presence stuff would be `undefined` and users could handle that however they want... 🤔

Reply:
I'm loath to add more config but maybe we could have something akin to a `strict` mode, which is `false` by default, but it can be set to `true` if someone wants the behaviour of "if any of the initial connection requests fail then I want the `catch` block to be hit of the connection promise".

Then we can have the default being that as long as the user subscription succeeds, we resolve the promise. WDYT?